app/queries/servers/thread.ts

```diff
@@ -52,6 +53,14 @@ export const observeThreadById = (database: Database, threadId: string) => {
     );
 };
 
+export const observeTeamIdByThreadId = (database: Database, threadId: string) => {
+    return database.get<ThreadInTeamModel>(THREADS_IN_TEAM).query(
+        Q.where('thread_id', threadId),
+    ).observe().pipe(
+        switchMap((threadsInTeam) => of$(threadsInTeam[0]?.teamId || undefined)),
```

Comment:
```suggestion
        switchMap((threadsInTeam) => of$(threadsInTeam[0]?.teamId)),
```
Should return you undefined already if the record does not exists